File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/TimeDurationValue.java
package jp.superwooo.chaipon.lighttimeswitcher;

public class TimeDurationValue {
    final private  int mMilliSecond;
    final private  int mSecond;
    public TimeDurationValue(int second, LimitTime limit){
        mSecond = limit.apply(second);
        mMilliSecond = mSecond * 1000;
    }

    @Override
    public boolean equals(Object value) {
        if(this == value) return true;
        if(value == null || getClass() != value.getClass()) return false;

        TimeDurationValue other = (TimeDurationValue)value;
        return mSecond == other.mSecond;
    }
    @Override
    public int hashCode(){
        return Integer.hashCode(mSecond);
    }



    public int sec(){return mSecond;}
    public int milliSecond(){return mMilliSecond;}

}

File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/TimeDurationValue.kt
package jp.superwooo.chaipon.lighttimeswitcher

class TimeDurationValue(second: Int, limit: LimitTime) {
    private val milliSecond: Int
    private val second = limit.apply(second)

    init {
        milliSecond = this.second * 1000
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val otherValue = other as TimeDurationValue
        return second == otherValue.second
    }

    override fun hashCode(): Int {
        return Integer.hashCode(second)
    }


    fun sec(): Int {
        return second
    }

    fun milliSecond(): Int {
        return milliSecond
    }
}
